content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
+
+- Understand Abstraction and Encapsulation
+
+- Create objects from classes
+
+- Have a working knowledge of what OOP is about
+
+- Create abstract classes and interfaces
+
+### CLASS
 
-###**CLASS**
 A class is a template from which an object is created. A class serves as a blueprint for object creation.
-Class has instance variables as attributes or properties and instance methods as their behaviors or actions
-Instance Variables: These are variables declared in the class, outside of methods and constructors. Instance variables have a global scope.
 
-###**Instance Methods**: This is a method that can be invoked only by an Object of its class.
+The Class has instance variables as attributes or properties and instance methods as their behaviors or actions
 
-It is the class that defines what an object data fields or variables and methods will be.
+Instance Variables: These are variables declared in the class, outside of methods and constructors. Instance variables have a global scope.
 
+### Instance Methods: 
 
+This is a method that can be invoked only by an Object of its class.
+
+It is the class that defines what an object data fields or variables and methods will be.
 
 Example of a class
 
 For instance, a circle class
-A circle has a radius from just this simple information we can create different real life objects of type circle with this attribute define by the Circle class and with a few action or methods.
+
+A circle has a radius from this, we can create different real-life objects of type circle. 
 
 ```java
+
 public class Circle {
-//instance variable
-  private double radius=2.5;
 
-//No argument constructor
-public Circle(){
+    //instance variable
 
-}
+ private double radius=2.5;
 
-//argument Constructor
+    //No argument constructor
 
-public Circle(double radius){
-this.radius = radius;
-}
+    public Circle(){
 
-//method
-public double getArea(){
-	return Math.PI*radius*radius; 
+    }
+
+    //argument Constructor
+
+    public Circle(double radius){
+
+    this.radius = radius;
 
     }
 
-//method
-public  double getPerimeter(){
-return Math.PI*(2*radius);
+    //method
+
+    public double getArea(){
+
+    return Math.PI*radius*radius; 
+
+    }
+
+    //method
+
+    public  double getPerimeter(){
+
+    return Math.PI*(2*radius);
+
+    }
 
-   }
 }
+
 ```
 
-With this, we’ve created a very simple class called Circle, the same goes for every other class like Car, Human, Animal, etc. Creating a class means you have a template or blueprints from which you can create different Objects of the same class.A class in simple term is a model. I hope you get this.
+With this, we’ve created a very simple class called Circle,
+The same goes for every other class like Car, Human, Animal, etc.
+Creating a class means you have a template or blueprints from which you can create different Objects of the same class. A class in simple terms is a model. I hope you get this.
+
+### OBJECTS
```

Comment:
No need to put the header *Objects* in all caps. Capitalizing it should be enough.